modules/core/src/test/java/org/apache/synapse/aspects/flow/statistics/publishing/PublishingEventTest.java

```diff
@@ -0,0 +1,117 @@
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.apache.synapse.aspects.flow.statistics.publishing;
+
+import junit.framework.Assert;
+import org.apache.synapse.aspects.ComponentType;
+import org.apache.synapse.aspects.flow.statistics.data.raw.StatisticDataUnit;
+import org.apache.synapse.aspects.flow.statistics.data.raw.StatisticsLog;
+import org.apache.synapse.aspects.flow.statistics.util.StatisticsConstants;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Unit tests for PublishingEvent class
+ */
+public class PublishingEventTest {
+
+    private static PublishingEvent publishingEvent;
+    private static final String COMPONENT_NAME = "testComponentName";
+    private static final String FLOW_ID = "testFlowId";
+    private static final String COMPONENT_ID = "testComponentId";
+    private static final Integer COMPONENT_INDEX = 1;
+    private static final Long START_TIME = 5L;
+    private static final Long END_TIME = 10L;
+    private static final Long DURATION = 5L;
+    private static final String BEFORE_PAYLOAD = "testBeforePayload";
+    private static final String AFTER_PAYLOAD = "testAfterPayload";
+    private static final Integer[] CHILDREN = {1, 2, 3};
+    private static final String ENTRY_POINT = "testEntryPoint";
+    private static final Integer ENTRY_POINT_HASH_CODE = 1234;
+    private static final Integer FAULT_COUNT = 1;
+    private static final Integer HASHCODE = 12345;
+    private static final Map<String, Object> PROPERTY_MAP = new HashMap<>();
+
+    /**
+     * Initializing PublishingEvent before tests
+     */
+    @BeforeClass
+    public static void init() {
+        StatisticDataUnit statisticDataUnit = new StatisticDataUnit();
+        statisticDataUnit.setTime(START_TIME);
+        PROPERTY_MAP.put("1", new Integer(1));
+        statisticDataUnit.setContextPropertyMap(PROPERTY_MAP);
+        StatisticsLog statisticsLog = new StatisticsLog(statisticDataUnit);
+        statisticsLog.setComponentType(ComponentType.ENDPOINT);
+        statisticsLog.setComponentName(COMPONENT_NAME);
+        statisticsLog.setComponentId(COMPONENT_ID);
+        statisticsLog.setEndTime(END_TIME);
+        statisticsLog.setChildren(Arrays.asList(CHILDREN));
+        statisticsLog.incrementNoOfFaults();
+        statisticsLog.setHashCode(HASHCODE);
+        publishingEvent = new PublishingEvent(FLOW_ID, COMPONENT_INDEX, statisticsLog, ENTRY_POINT,
+                ENTRY_POINT_HASH_CODE);
+        publishingEvent.setBeforePayload(BEFORE_PAYLOAD);
+        publishingEvent.setAfterPayload(AFTER_PAYLOAD);
+    }
+
+    /**
+     * Testing the overridden toSting method
+     */
+    @Test
+    public void testToString() {
+        String expected = "Component Type " + StatisticsConstants.FLOW_STATISTICS_ENDPOINT +
```

Comment:
line wrapping should happen before an operator. Shall we move the '+' sign to the next line?